server/engine/src/main/java/org/eclipse/lsp/cobol/lsp/analysis/AsyncAnalysisService.java

```diff
@@ -300,19 +299,20 @@ public void reanalyseCopybooksAssociatedPrograms(
     for (String uri : openedUris) {
       String languageId = documentModelService.get(uri).getLanguageId();
       // TODO: update cache directly from workspace document graph
-      copybookService.getCopybookUsage(uri).stream()
-          .filter(model -> Objects.nonNull(model.getUri()))
-          .filter(model -> model.getUri().equals(copybookUri))
-          .forEach(
-              copybookModel -> {
-                copybookService.invalidateCache(copybookModel.getCopybookId());
-                if (copybookContent != null) {
-                  copybookModel.setContent(copybookContent);
-                  copybookService.store(
-                      copybookModel,
-                      dialectService.getPreprocessor(CobolLanguageId.MAPPER.get(languageId)));
-                }
-              });
+      //      copybookService.getCopybookUsage(uri).stream()
+      //          .filter(model -> Objects.nonNull(model.getUri()))
+      //          .filter(model -> model.getUri().equals(copybookUri))
+      //          .forEach(
+      //              copybookModel -> {
+      //                copybookService.invalidateCache(copybookModel.getCopybookId());
+      //                if (copybookContent != null) {
+      //                  copybookModel.setContent(copybookContent);
+      //                  copybookService.store(
+      //                      copybookModel,
+      //
+      // dialectService.getPreprocessor(CobolLanguageId.MAPPER.get(languageId)));
+      //                }
+      //              });
```

Comment:
Needs testing, but should not be required now as the copybook content in not cached , only the uri resolution is cached